crates/solidity/inputs/schema/grammar/topics/06-yul/03-yul-expressions.yml

```diff
@@ -4,10 +4,27 @@
   kind: "Rule"
   unversioned:
     choice:
-      - reference: "YulIdentifierPath"
       - reference: "YulFunctionCall"
       - reference: "YulLiteral"
 
+- name: "YulFunctionCall"
+  kind: "Rule"
+  unversioned:
+    sequence:
+      - reference: "YulIdentifierPath"
+      # TODO: temporary hack until precedence is implemented properly here.
```

Comment:
Noting the temporary hack until precedence is redesigned/fixed.

Reply:
Precedence is working AFAIK, so can you explain what the issue is here?

Reply:
This is the failure I mentioned earlier. The current PR is a hack to make parsing succeed, while we investigate next.

Below is the precedence alternative. It currently fails with `A TerminalTrie with no entries should never have been constructed`. It assumes the existence of a binary operator, and fails as such. Might also be a part of our discussion on enforcing the explicit structure of each operator type (ternary/binary/unary) via YAML (an EBNF variant) instead detecting it later.

```yml
- name: "YulExpression"
  kind: "Rule"
  unversioned:
    parserType: "Precedence"
    choice:
      - reference: "YulFunctionCall"
      - reference: "YulPrimaryExpression"

- name: "YulFunctionCall"
  kind: "Rule"
  unversioned:
    sequence:
      - reference: "YulExpression"
      - delimitedBy:
          open: "("
          expression:
            optional:
              separatedBy:
                separator: ","
                expression:
                  reference: "YulExpression"
          close: ")"

- name: "YulPrimaryExpression"
  kind: "Rule"
  unversioned:
    choice:
      - reference: "YulIdentifierPath"
      - reference: "YulLiteral"
```

Reply:
Ah, I get it. easy to fix. I'll approve and schedule a fix.